exec/pipeline/financials/structs.py

```diff
@@ -0,0 +1,140 @@
+from pydantic import BaseModel, Field, validator
+from decimal import Decimal
+
+
+class NetSales(BaseModel):
+    producs: Decimal
+    services: Decimal
+    total_net_sales: Decimal
+
+class CostofSales(BaseModel):
+    products: Decimal
+    services: Decimal
+    total_cost_of_sales: Decimal
+
+
+
+class OperatingExpenses(BaseModel):
+    research_and_development: Decimal
+    selling_general_and_administrative: Decimal
+    total__operating_expenses: Decimal
+
+
+class CurrentAssets(BaseModel):
+    cash_and_cash_equivalents: Decimal
+    marketable_securities: Decimal
+    accounts_receivable_net: Decimal
+    vendor_non_trade_receivables: Decimal
+    other_current_assets: Decimal
+    total_current_assets: Decimal
+
+    # @validator("total_current_assets")
+    # def total_current_assets_is_sum_of_all_assets(cls, v):
+    #     return v == (
+    #         cls.cash_and_cash_equivalents
+    #         + cls.marketable_securities
+    #         + cls.accounts_receivable_net
+    #         + cls.vendor_non_trade_receivables
+    #         + cls.other_current_assets
+    #     )
+
+
+
+
+class NonCurrentAssets(BaseModel):
+    marketable_securities: Decimal
+    property_plant_and_equipment_net: Decimal
+    other_non_current_assets: Decimal
+    total_non_current_assets: Decimal
+
+
+class Assets(BaseModel):
+    current_assets: CurrentAssets
+    non_current_assets: NonCurrentAssets
+
+
+class EarningsPerShare(BaseModel):
+    basic: Decimal
+    diluted: Decimal
+
+class Shares(BaseModel):
+    basic: int
+    diluted: int
+
+
+
+class NetSalesByReportableSegment(BaseModel):
+    americas: Decimal
+    europe: Decimal
+    greater_china: Decimal
+    japan: Decimal
+    rest_of_asia_pacific: Decimal
+    total_net_sales: Decimal
+
+
+class NetSalesByCategory(BaseModel):
+    iphone: Decimal
+    mac: Decimal
+    ipad: Decimal
+    wearable_home_and_accessories: Decimal
+    services: Decimal
+    total_net_sales: Decimal
```

Comment:
Go generalize this I'd imagine it'd need to be a dict or something.

Reply:
yea probably, just have it be a generic dict across companies, let's see how it performs upstack on more data, see what works